### PR TITLE
Add Kokoro bots for building with Bazel.

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

File: kokoro/linux-clang-release-bazel/build.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# Display commands being run.
+set -x
+
+CC=clang
+CXX=clang++
+
+BUILD_ROOT=$PWD
+SRC=$PWD/github/SPIRV-Tools
+cd $SRC
+git clone --depth=1 https://github.com/KhronosGroup/SPIRV-Headers external/spirv-headers
```
Comment (dj2) [inline kokoro/linux-clang-release-bazel/build.sh]: utils/git-sync-deps

(same for other scripts)

File: kokoro/linux-clang-release-bazel/build.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+git clone --depth=1 https://github.com/google/googletest          external/googletest
+git clone --depth=1 https://github.com/google/effcee              external/effcee
+git clone --depth=1 https://github.com/google/re2                 external/re2
+
+gsutil cp gs://bazel/0.29.1/release/bazel-0.29.1-linux-x86_64 .
+chmod +x bazel-0.29.1-linux-x86_64
+alias bazel=bazel-0.29.1-linux-x86_64
+bazel --version
+
+bazel build :all
```
Comment (dj2) [inline kokoro/linux-clang-release-bazel/build.sh]: Maybe add an echo ${date} starting build and echo ${date} starting tests similar to other scripts?

File: kokoro/linux-clang-release-bazel/build.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+set -e
+# Display commands being run.
+set -x
+
+CC=clang
+CXX=clang++
+SRC=$PWD/github/SPIRV-Tools
+
+cd $SRC
+./utils/git-sync-deps
```
Comment (s-perron) [inline kokoro/linux-clang-release-bazel/build.sh]: Our bots use the tip-of-tree for all of our dependencies.  We considered changing it (see https://github.com/KhronosGroup/SPIRV-Tools/pull/2345), but we need an autoroller first.  We need to be sure the deps file does not fall too far behind.

I know Dan asked for this, but I think we should do what the other bots do.

File: kokoro/macos-clang-release-bazel/build.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+set -e
+# Display commands being run.
+set -x
+
+CC=clang
+CXX=clang++
+SRC=$PWD/github/SPIRV-Tools
+
+cd $SRC
+./utils/git-sync-deps
```
Comment (s-perron) [inline kokoro/macos-clang-release-bazel/build.sh]: Same as above.

File: kokoro/windows-msvc-2017-release-bazel/build.bat [hunk 5]
```diff
@@ -0,0 +1,73 @@
+:: Copyright (c) 2019 Google LLC.
```
Comment (s-perron) [inline kokoro/windows-msvc-2017-release-bazel/build.bat]: Might need to rename the file if you end up using MSVC 2015.

File: kokoro/linux-clang-release-bazel/build.sh (hunk 3 above)
Comment (ehsannas) [reply kokoro/linux-clang-release-bazel/build.sh]: Done

File: kokoro/macos-clang-release-bazel/build.sh (hunk 4 above)
Comment (ehsannas) [reply kokoro/macos-clang-release-bazel/build.sh]: Done

File: kokoro/windows-msvc-2017-release-bazel/build.bat (hunk 5 above)
Comment (ehsannas) [reply kokoro/windows-msvc-2017-release-bazel/build.bat]: Done

File: kokoro/linux-clang-release-bazel/build.sh (hunk 1 above)
Comment (s-perron) [reply kokoro/linux-clang-release-bazel/build.sh]: We don't have a process for keeping the deps file up-to-date.  We should just use tip-of-tree for each of the dependencies.

**Comment by ehsannas** [comment]
(╯°□°)╯︵ ┻━┻

**Comment by ehsannas** [comment]
soooo now the tests fail :/

File: kokoro/linux-clang-release-bazel/build.sh [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+git clone --depth=1 https://github.com/google/re2                 external/re2
+
+gsutil cp gs://bazel/0.29.1/release/bazel-0.29.1-linux-x86_64 .
+chmod +x bazel-0.29.1-linux-x86_64
+
+echo $(date): Build everything...
+./bazel-0.29.1-linux-x86_64 build :all
+echo $(date): Build completed.
+
+echo $(date): Starting ctest...
```
[the remark this answers is not in the document]
Comment (ehsannas) [reply kokoro/linux-clang-release-bazel/build.sh]: Done

File: kokoro/macos-clang-release-bazel/build.sh [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+# Get bazel 0.29.1.
+gsutil cp gs://bazel/0.29.1/release/bazel-0.29.1-darwin-x86_64 .
+chmod +x bazel-0.29.1-darwin-x86_64
+
+echo $(date): Build everything...
+./bazel-0.29.1-darwin-x86_64 build :all
+echo $(date): Build completed.
+
+echo $(date): Starting ctest...
```
[the remark this answers is not in the document]
Comment (ehsannas) [reply kokoro/macos-clang-release-bazel/build.sh]: Done

**Comment by ehsannas** [comment]
PTAL

**Pull request merged**